content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,143 @@
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+ name: basic-auth
+ namespace: istio-system
+spec:
+ configPatches:
+ - applyTo: HTTP_FILTER
+   match:
+     context: GATEWAY
+     listener:
+       filterChain:
+         filter:
+           name: envoy.http_connection_manager
+   patch:
+     operation: INSERT_BEFORE
+     value:
+       name: istio.basic_auth
+       config_discovery:
+         config_source:
+           ads: {}
+           initial_fetch_timeout: 0s # wait indefinitely to prevent bad Wasm fetch
+         type_urls: [ "type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm"]
+{{< /text >}}
+
+The second `EnvoyFilter` provides configuration for the filter, which is an `EXTENSION_CONFIG` patch and will be distributed to the proxy as an Envoy [`Extension Configuration`](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS) resource.
+Once this update reaches the Istio agent, the agent will download the Wasm module and store it in the local file system.
+If the download fails, the agent will reject the `Extension Configuration` update to prevent invalid Wasm filter configuration from reaching the Envoy proxy.
+Because of this protection, with the initial fetch timeout being set to 0, the listener update will not become effective and an invalid Wasm filter will not disturb traffic.
+The important parts of this configuration are:
+
+- Wasm `vm` configuration which points to a remote Wasm module.
+- Wasm extension configuration, which is a JSON string that is consumed by the Wasm extension.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+ name: basic-auth-config
+ namespace: istio-system
+spec:
+ configPatches:
+ - applyTo: EXTENSION_CONFIG
+   match:
+     context: GATEWAY
+   patch:
+     operation: ADD
+     value:
+       name: istio.basic_auth
+       typed_config:
+         "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+         type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
+         value:
+           config:
+             vm_config:
+               vm_id: basic-auth
+               runtime: envoy.wasm.runtime.v8
+               code:
+                 remote:
+                   http_uri:
+                     uri: https://github.com/istio-ecosystem/wasm-extensions/releases/download/{{< istio_version >}}.0/basic-auth.wasm
```

Comment:
should we maybe add this parametrized URL as a comment and use the `.wasm` binary for 1.9 here? we can scope the EnvoyFilter to proxies with version 1.8. That would allow us to use the sha256 as well and establish it as a best practice

Reply:
I am trying to avoid remembering updating this doc for every release, since it is really easy to forget. I've update the comment to make it clear that checksum is highly recommended.